R/tm_g_distribution.R

```diff
@@ -1198,52 +1264,88 @@ srv_distribution <- function(id,
             )
           )
         }
-        qenv %>%
-          # used to display table when running show-r-code code
-          teal.code::eval_code(quote(test_stats))
       }
     )
 
     # outputs ----
     ## building main qenv
-    output_q <- reactive({
-      tab <- input$tabs
-      req(tab) # tab is NULL upon app launch, hence will crash without this statement
-
-      qenv_final <- common_q()
+    output_common_q <- reactive({
       # wrapped in if since could lead into validate error - we do want to continue
-      test_r_qenv_out <- try(test_q(), silent = TRUE)
-      if (!inherits(test_r_qenv_out, c("try-error", "error"))) {
-        qenv_final <- c(qenv_final, test_q())
+      test_q_out <- try(test_q(), silent = TRUE)
+      if (!inherits(test_q_out, c("try-error", "error"))) {
+        c(
+          common_q(),
+          within(test_q_out, {
+            test_table <- DT::datatable(
+              test_table_data,
+              options = list(scrollX = TRUE),
+              rownames = FALSE
+            )
+          })
+        )
+      } else {
+        within(common_q(), test_table <- NULL)
       }
+    })
+
+    output_dist_q <- reactive(c(output_common_q(), req(dist_q())))
+    output_qq_q <- reactive(c(output_common_q(), req(qq_q())))
+
+    decorated_output_dist_q <- srv_decorate_teal_data(
+      "d_density",
+      data = output_dist_q,
+      decorators = select_decorators(decorators, "histogram_plot"),
+      expr = print(histogram_plot)
+    )
+
+    decorated_output_qq_q <- srv_decorate_teal_data(
+      "d_qq",
+      data = output_qq_q,
+      decorators = select_decorators(decorators, "qq_plot"),
+      expr = print(qq_plot)
+    )
+
+    decorated_output_summary_q <- srv_decorate_teal_data(
+      "d_summary",
+      data = output_common_q,
+      decorators = select_decorators(decorators, "summary_table"),
+      expr = summary_table
+    )
+
+    decorated_output_test_q <- srv_decorate_teal_data(
+      "d_test",
+      data = output_common_q,
+      decorators = select_decorators(decorators, "test_table"),
+      expr = test_table
+    )
 
-      qenv_final <- if (tab == "Histogram") {
-        req(dist_q())
-        c(qenv_final, dist_q())
-      } else if (tab == "QQplot") {
-        req(qq_q())
-        c(qenv_final, qq_q())
+    decorated_output_q <- reactive({
```

Comment:
Changed here to your suggestion @m7pr build each output individually and then use `c.qenv()` to merge them.

Reply:
The logic here is a bit different as the module was fault-tolerant to test_table having missing input error

Reply:
Alrighty @averissimo . I will also incorporate in tmc cases like in here https://github.com/insightsengineering/teal.modules.clinical/pull/1267#discussion_r1862381325